system.go

```diff
@@ -1,12 +1,43 @@
 package nitriding
 
-import "syscall"
+import (
+	"errors"
+	"io"
+	"syscall"
+)
 
 const (
-	defaultFdCur = 65536
-	defaultFdMax = 65536
+	defaultFdCur     = 65536
+	defaultFdMax     = 65536
+	errTooMuchToRead = "reached read limit"
 )
 
+// limitReader behaves like a Reader but it returns errTooMuchToRead if the
+// read limit was met or exceeded.
+type limitReader struct {
+	io.Reader
+	Limit int
+}
```

Comment:
I see there's an `io.LimitedReader` if that's better than implementing it yourself.

Reply:
That's what I originally thought, but `LimitedReader` does not distinguish between EOF and "too many bytes to read", which is important for us.

Reply:
I see, thanks. I saw a 3rd-party package which claimed to address that very issue earlier, but can't find it now. It's not much code, so having a separate impl is fine.

A std library fix for this [almost made it into go 1.19](https://github.com/golang/go/issues/51115#issuecomment-1146642811). Maybe next release!

Reply:
Ah, that's good detective work.